editor/file_system/editor_file_system.cpp

```diff
@@ -2559,8 +2559,8 @@ void EditorFileSystem::_register_global_class_script(const String &p_search_path
 	if (lang.is_empty()) {
 		return; // No lang found that can handle this global class
 	}
-
-	ScriptServer::add_global_class(p_script_update.name, p_script_update.extends, lang, p_target_path, p_script_update.is_abstract, p_script_update.is_tool);
+	ResourceUID::ID uid = ResourceLoader::get_resource_uid(p_target_path);
```

Comment:
```suggestion
	ResourceUID::ID uid = get_file_uid(p_target_path);
```
This is more efficient, but will only work if filesystem cache exists. Not sure if that's true here.